File: scala/src/main/scala/services.scala
package com.jambit.practicalresult

type Service[T] = (t: T) => Either[String, T]

val byTwo: Service[Int] = i => Right(i * 2)

val plusOneIfEven: Service[Int] =
  i => if (i % 2 != 0) Left("I don't like odds!")
       else Right(i + 1)

File: kotlin/src/main/java/Services.kt
import arrow.core.Either

typealias MapService<T> = (t: T) -> Either<String, T>

val byTwo: MapService<Int> = { i -> Either.Right(i * 2) }

val plusOneIfEven: MapService<Int> = { i ->
    if (i % 2 != 0) Either.Left("I don't like odds!")
    else Either.Right(i + 1)
}